Packs/ExabeamDataLake/Integrations/ExabeamDataLake/ExabeamDataLake.py

```diff
@@ -0,0 +1,273 @@
+
+    Args:
+        entry: The entry from the API response.
+
+    Returns:
+        dict: The parsed entry dictionary.
+    """
+    source: dict = entry.get("_source", {})
+    return {
+        "Id": entry.get("_id"),
+        "Vendor": source.get("Vendor"),
+        "Created_at": source.get("@timestamp"),
+        "Product": source.get("Product"),
+        "Message": source.get("message")
+    }
+
+
+def dates_in_range(start_time: Any, end_time: Any) -> list[str]:
+    """
+     Generate a list of dates within a specified range.
+
+    Args:
+        start_time: The start date of the range in the format "YYYY-MM-DD".
+        end_time: The end date of the range in the format "YYYY-MM-DD".
+
+    Raises:
+        DemistoException: If the start time is not before the end time, or if the difference between start time and end time is
+        greater than 10 days.
+
+    Returns:
+        list: A list of dates within the specified range, formatted as strings in the format "YYYY.MM.DD".
+    """
+    start_time = datetime.strptime(start_time, "%Y-%m-%d")
+    end_time = datetime.strptime(end_time, "%Y-%m-%d")
+
+    if start_time >= end_time:
+        raise DemistoException("Start time must be before end time")
+
+    if (end_time - start_time).days > 10:
+        raise DemistoException("Difference between start time and end time must be less than or equal to 10 days")
+
+    dates = []
+    current_date = start_time
+    while current_date <= end_time:
+        dates.append(current_date.strftime("%Y.%m.%d"))
+        current_date += timedelta(days=1)
+
+    return dates
+
+
+def get_date(time: str):
+    """
+    Get the date from a given time string.
+
+    Args:
+        time (str): The time string to extract the date from.
+
+    Returns:
+        str: The date extracted from the time string formatted in ISO 8601 format (YYYY-MM-DD),
+        or None if the time string is invalid.
+    """
+    date_time = arg_to_datetime(arg=time, arg_name="Start time", required=True)
+    if date_time:
+        date = date_time.strftime(ISO_8601_FORMAT)
+    return date
+
+
+def calculate_page_parameters(args: dict) -> tuple[int, int]:
+    """
+      Calculate the page parameters for pagination.
+
+    Args:
+        args: A dictionary containing the arguments passed to the function.
+
+    Raises:
+        DemistoException: If invalid combinations of arguments are provided. You can only provide 'limit'
+        alone or 'page' and 'page_size' together.
+
+    Returns:
+        tuple: A tuple containing two integers representing the 'from' and 'size' parameters for pagination.
+        'from' is the index of the first item to retrieve, and 'size' is the number of items to retrieve.
+    """
+    page_arg = args.get('page')
+    page_size_arg = args.get('page_size')
+    limit_arg = args.get('limit')
+
+    if (limit_arg and (page_arg or page_size_arg)) or ((not (page_arg and page_size_arg)) and (page_arg or page_size_arg)):
```

Comment:
This line is not very redable and also has a bug, please check the scenario that all parameter arguments are none your function will not enter the if statment. And will not raise the exception.
```suggestion
    if (limit_arg and (page_arg or page_size_arg) or (not limit_arg and not (page_arg and page_size_arg))):
```